### PR TITLE
Require that the customizations of get_env are noexcept.

### LINKED ISSUE #430: get_env should be noexcept
`execution::get_env` should be specified to be `noexcept`, as this is a fundamental building block function for all the traits queries.
It should thus mandate that the `tag_invoke` expression is `noexcept` if it is customized.

### PR DESCRIPTION
**Pull request by griwes**
Resolves #430.

### CODE REVIEW

**Review by miscco — APPROVED** [review]

File: std_execution.bs [hunk 1]
```diff
@@ -4050,7 +4050,7 @@ namespace std::execution {
 
     1. `tag_invoke(execution::get_env, r)` if that expression is well-formed.
 
-        * <i>Mandates:</i> The decayed type of the above expression is not `no_env`.
+        * <i>Mandates:</i> The decayed type of the above expression is not `no_env` and `noexcept(tag_invoke(execution::get_env, r))` is `true`.
```
Comment (ericniebler) [inline std_execution.bs:4053]: Elsewhere we use the phrase: "and the expression XXX is not potentially throwing."

Comment (griwes) [reply std_execution.bs:4053]: How about "The above expression is not potentially throwing, and its decayed type is not `no_env`."?

Comment (griwes) [reply std_execution.bs]: Done.

**Comment by ericniebler** [comment]
As noted in #430, this has implementation implications. My preference would be for this PR to come with associated changes to the reference implementation, but I would be satisfied with a new issue for that so we don't lose track of it.

**Comment by ericniebler** [comment]
This is now brycelelbach#1

**Pull request closed without merging**